app/src/main/java/com/github/freeman/bootcamp/games/guessit/lobbies/CreatePublicPrivateActivity.kt

```diff
@@ -0,0 +1,133 @@
+import com.github.freeman.bootcamp.games.guessit.lobbies.CreatePublicPrivateActivity.Companion.PUBLIC_BUTTON_TEXT
+import com.github.freeman.bootcamp.games.guessit.lobbies.CreatePublicPrivateActivity.Companion.PUBLIC_TYPE_TEXT
+import com.github.freeman.bootcamp.games.guessit.lobbies.CreatePublicPrivateActivity.Companion.TOPBAR_TEXT
+import com.github.freeman.bootcamp.games.guessit.lobbies.CreatePublicPrivateActivity.Companion.TYPE_TEXT
+import com.github.freeman.bootcamp.ui.theme.BootcampComposeTheme
+
+/**
+ * Shows a screen where you can either choose to create a public or a private lobby
+ */
+class CreatePublicPrivateActivity: ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            BootcampComposeTheme {
+                TopAppbarPublicPrivate()
+
+                MainScreen()
+            }
+        }
+    }
+
+    companion object {
+        const val TOPBAR_TEXT = "Create a lobby"
+        const val PUBLIC_BUTTON_TEXT = "Public lobby"
+        const val PRIVATE_BUTTON_TEXT = "Private lobby"
+        const val TYPE_TEXT = "type"
+        const val PUBLIC_TYPE_TEXT = "public"
+        const val PRIVATE_TYPE_TEXT = "private"
+    }
+}
+
+
+
+@Composable
+fun TopAppbarPublicPrivate(context: Context = LocalContext.current) {
+
+    TopAppBar(
+        modifier = Modifier.testTag("topAppbarPublicPrivate"),
+        title = {
+            Text(
+                text = TOPBAR_TEXT,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
+            )
+        },
+        backgroundColor = MaterialTheme.colors.background,
+        elevation = 4.dp,
+        navigationIcon = {
+            IconButton(
+                modifier = Modifier.testTag("topAppbarPublicPrivateButton"),
+                onClick = {
+                val activity = (context as? Activity)
+                activity?.finish()
+            }) {
+                Icon(
+                    Icons.Filled.ArrowBack,
+                    contentDescription = "Go back",
+                )
+            }
+        }
+    )
+}
+
+@Composable
+fun MainScreen() {
+    Column (
+        modifier = Modifier
+            .fillMaxSize()
+            .testTag("createPublicPrivateMainScreen"),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ){
+        PublicLobbyButton()
+        Spacer(modifier = Modifier.size(6.dp))
+        PrivateLobbyButton()
+    }
+}
+
+@Composable
+private fun PublicLobbyButton() {
+    val context = LocalContext.current
+    ElevatedButton(
+        modifier = Modifier.testTag("publicLobbyButton"),
+        onClick = {
+            context.startActivity(
+                Intent(context, GameOptionsActivity::class.java)
+                    .putExtra(TYPE_TEXT, PUBLIC_TYPE_TEXT)
```

Comment:
As suggested in another one of my comments, maybe the TYPE_TEXT String could be replaced by the "type_extra" String in strings.xml.